walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -114,4 +145,56 @@
 
     return possibleSlots;
   }
+
+  /**
+   * Find the intersection of two collections of slots. The intersection between two individual
+   * slots is the slot that is contained within the other one. If none of the two slots is contained
+   * within the other one, then their intersection is empty.
+   *
+   * @param mandatory A collection of {@code TimeRange}s that represents meeting slots for mandatory
+   *     attendees
+   * @param optional A collection of {@code TimeRange}s that represents meeting slots for optional
+   *     attendees
+   * @return The intersection of these collections as specified above.
+   */
+  public Collection<TimeRange> intersection(
```

Comment:
I would rename this to something like calculateOverallAvailableSlots() (or along those lines) for clarity. The name `intersection` with this method signature implies a standard set intersection i.e. what exact TimeRange objects are in both sets. That's not quite what we're doing here.